test/unit/visitor/sympy_solver.cpp

```diff
@@ -64,6 +64,92 @@ std::vector<std::string> run_sympy_solver_visitor(
     return results;
 }
 
+void compare_blocks(const std::string& result, const std::string& expected) {
+    using namespace pybind11::literals;
+
+    auto locals = pybind11::dict("s1"_a = result, "s2"_a = expected, "is_equal"_a = false);
+    pybind11::exec(R"(
+                    def compare_blocks(s1, s2):
+                        def sanitize(s):
+                            import re
+                            d = {'\[(\d+)\]':'_\\1',  'pow\((\w+), ?(\d+)\)':'\\1**\\2'}
+                            out = s
+                            for key, val in d.items():
+                                out = re.sub(key, val, out)
+                            return out
+
+                        def compare_systems_of_eq(d1, d2):
+                            from sympy.parsing.sympy_parser import parse_expr
+                            try:
+                                for k, v in d1.items():
+                                    if parse_expr(f'simplify(({v})-({d2[k]}))'):
+                                        return False
+                            except KeyError:
+                                return False
+
+                            d1.clear()
+                            d2.clear()
+                            return True
+
+                        def reduce(s):
```

Comment:
some docstrings would be nice, to explain bit of logic

Reply:
done! However, in clion they will not appear highlghted (because it is python code inside a string). I also added a doxygen explaination for the whole funciton